bazel/bazel.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -e
+
+# Bazel Wrapper script algorithm:
```

Comment:
Is there any chance we could convince Bazel to adopt this algorithm itself, and remove the need for this wrapper? Or is there something that Bazel already supports that we can piggyback off of? Injecting our own wrapper script isn't the worst thing in the world (we do it for `kubectl`) but I'd like to avoid it long-term if possible.

For `kubectl` we also retain the actual `kubectl` binary on the `PATH`, and only inject the wrapper in the builder's `ENTRYPOINT` -- this has the downside that users who do something like:

```yaml
- name: gcr.io/cloud-builders/kubectl
  entrypoint: bash
  args: ['-c', 'kubectl apply -f foo.yaml']
```

will bypass the wrapper, probably unknowingly. I'm not sure whether replacing the binary with the wrapper is more or less expected behavior.